test/actor_test.py

```diff
@@ -2138,3 +2139,44 @@ def method(self):
         ray.wait([object_id])
 
     ray.get(results)
+
+@pytest.mark.skipif(
```

Comment:
Can you also add a test that makes sure that an actor that was intentionally killed (e.g., because a `__ray_terminate__` task was submitted) does not get reconstructed, even if `max_reconstructions > 0`?